### PR TITLE
Don't create scaling vector when we aren't doing AD

### PR DESCRIPTION
**Pull request by lindsayad**
With the merge of #16440, global AD became our default config. Griffin
revealed some undesirable behavior in the global AD code-base
post-merge. We were creating the `NumericVector` of scaling factors
anytime that any variable had a non-unity scaling factor; however, this
vector is actually only ever used through AD residual/jacobian computing
objects. There is another issue which is that the
`assembleScalingVector` code does not yet work with array variables,
which is the reason Griffin failed: we were trying to assemble this
unnecessary vector, and that code triggered the assertion about not
working with array variables. https://github.com/idaholab/moose/issues/16820 created for that.
In the meantime (and in the future) we will not build the scaling vector
if we do not have AD objects

### CODE REVIEW

**Comment by lindsayad** [comment]
@YaqiWang this patch fixed the Griffin assertions for me. I will add a simple MOOSE test here

**Comment by YaqiWang** [comment]
Thanks Alex.

**Review by GiudGiud — COMMENTED** [review]
dug down the doxygen to see what all the setup that were called are and why but someone with more experience on the setup should review.

File: framework/src/systems/NonlinearSystemBase.C [hunk 1]
```diff
@@ -3546,7 +3563,7 @@ NonlinearSystemBase::assembleScalingVector()
   for (const auto * const scalar_var : scalar_variables)
   {
     mooseAssert(scalar_var->count() == 1,
-                "Contact Robert Carlsen and tell him to make this work for array variables.");
+                "Contact Alex Lindsay and tell him to make this work for array variables.");
```
[the remark this answers is not in the document]
Comment (lindsayad) [reply framework/src/systems/NonlinearSystemBase.C:3566]: :smile:

**Review by rwcarlsen — APPROVED** [review]

**Pull request merged**